python/lsst/ts/tunablelaser/component.py

```diff
@@ -138,12 +157,30 @@ async def set_burst_count(self, count):
             The amount to pulse the laser.
         """
         await self.m_cpu800.set_burst_count(count)
+        await self.csc.evt_burstCountSet.set_write(count=count)
 
     async def start_propagating(self):
         """Start propagating the beam of the laser."""
         await self.m_cpu800.start_propagating()
+        await asyncio.sleep(10)  # laser warmup delay
+        if (
+            self.m_cpu800.continous_burst_mode_trigger_burst_register.register_value
+            == Mode.BURST
+        ):
+            await self.csc.publish_new_detailed_state(
+                LaserDetailedState.PROPAGATINGBURSTMODEWAITINGFORTRIGGER
+            )
+        elif (
+            self.m_cpu800.continous_burst_mode_trigger_burst_register.register_value
+            == Mode.CONTINUOUS
+        ):
+            await self.csc.publish_new_detailed_state(LaserDetailedState.PROPAGATING)
         self.is_propgating = True
 
+    async def start_trigger(self):
+        """Trigger the laser's burst."""
+        await self.set_trigger_mode()
```

Comment:
Having two methods that do the same thing is confusing. I suggest you delete this and look at the comments on set_trigger_mode above.